rclcpp/test/rclcpp/test_reinitialized_timers.cpp

```diff
@@ -0,0 +1,108 @@
+// Copyright 2024 iRobot Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+
+#include "rclcpp/experimental/executors/events_executor/events_executor.hpp"
+#include "rclcpp/rclcpp.hpp"
+
+class TestTimersLifecycle
+: public testing::Test
+{
+public:
+  void SetUp() override
+  {
+    rclcpp::init(0, nullptr);
+  }
+
+  void TearDown() override
+  {
+    rclcpp::shutdown();
+  }
+};
+
+TEST_F(TestTimersLifecycle, timers_lifecycle_reinitialized_object)
```

Comment:
Should this test be run on all executors?
Even if the bug was found only in the events-executor, what we are verifying is the expected behavior for all executors

Reply:
We should also add a comment describing more in details the issues that resulted in this test.
For example copy-pasting here the PR description

Reply:
Yes, agreed with all of what @alsora said.  I also think the test should move into the `executors` subdirectory.

Reply:
Comments from both of you guys were addrressed